great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -712,6 +723,71 @@ def build_batch_request(
             partitioner=partitioner,
         )
 
+    @override
+    def add_batch_definition(
+        self,
+        name: str,
+        partitioner: Optional[ColumnPartitioner] = None,
+    ) -> BatchDefinition[ColumnPartitioner]:
+        if partitioner:
+            self.validate_batch_definition(partitioner)
+        return super().add_batch_definition(name, partitioner)
+
+    @public_api
+    def validate_batch_definition(self, partitioner: ColumnPartitioner) -> None:
+        """Validates that the batch definition column is of a permissible type
+
+        This isn't meant to be called directly. This is called internally when a batch definition
+         is added. Data asset implementers can override this for their specific data asset.
+
+        Raises:
+            SqlAddBatchDefinitionError: The specified column to partition on is not of
+                a permissible type for batching (ie date or datetime) or no data is
+                present in this column.
+        """
+        # We only support certain partitioners for using as batch definitions.
```

Comment:
I think this class's types need to be overhauled in terms of partitioners since we only support a subset how the classes typing is setup with generics would make this a big change which I didn't want to take on atm.